app/controllers/api/users.php

```diff
@@ -92,7 +102,18 @@ function createUser(string $hash, mixed $hashOptions, string $userId, ?string $e
             }
         }
 
-        $password = (!empty($password)) ? ($hash === 'plaintext' ? Auth::passwordHash($password, $hash, $hashOptionsObject) : $password) : null;
+        $hashedPassword = null;
+
+        if (!empty($password)) {
+            if ($hash instanceof Plaintext) { // Password was never hashed, hash it with the default hash
+                $defaultHash = new ProofsPassword();
+                $hashedPassword = $defaultHash->hash($password);
+                $hash = $defaultHash->getHash();
```

Comment:
This is probably okay for now, but if shutdown hook wanted to look at hash, he would not see those changes; might become problem for coroutines (original value in resource does not get overriden)

Lets add todo at least, so we can find this place easily in future

Reply:
alternatively, methods like setAlgo, setOptions, ... could work

Or maybe like we have in events, functon to from(), reset()